app/models/user.py

```diff
@@ -46,7 +47,14 @@ def isBonnerScholar(self):
             self._bsCache = BonnerCohort.select().where(BonnerCohort.user == self).exists()
 
         return self._bsCache
-
+    
+    @property
+    def hasCurrentCeltsLabor(self):
+        if self._bsCache is None:
+            from app.models.celtsLabor import CeltsLabor
+            self._bsCache = CeltsLabor.select().where(CeltsLabor.user == self, CeltsLabor.term.isCurrent == True).exists()
```

Comment:
This query fails because the term table isn't being included, so it can't check `isCurrent`. You need to join with Term